### PR TITLE
[Impeller] ensure 1x1 has mipcount of 1

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/119245

Before mipcount of 1x1 image would be 0, resulting in invalid texture

### CODE REVIEW

**Comment by jonahwilliams** [comment]
actually I think I might be misunderstanding this API, certainly asking for a mipcount of 0 is invalid - but does mipcount of 1 mean no mipping mapping?

File: impeller/geometry/size.h [hunk 1]
```diff
@@ -102,7 +102,9 @@ struct TSize {
     if (!IsPositive()) {
       return 1u;
     }
-    return std::max(ceil(log2(width)), ceil(log2(height)));
+    constexpr size_t minimumMip = 1u;
```
Comment (chinmaygarde) [inline impeller/geometry/size.h:105]: underscore_case

File: impeller/geometry/geometry_unittests.cc [hunk 2]
```diff
@@ -640,6 +640,7 @@ TEST(GeometryTest, CanGenerateMipCounts) {
   ASSERT_EQ((Size{128, 0}.MipCount()), 1u);
   ASSERT_EQ((Size{128, -25}.MipCount()), 1u);
   ASSERT_EQ((Size{-128, 25}.MipCount()), 1u);
+  ASSERT_EQ((Size{1, 1}.MipCount()), 1u);
```
Comment (chinmaygarde) [inline impeller/geometry/geometry_unittests.cc:643]: Mip counts should always at least be 1. Perhaps also check `Size{0, 0}`?

**Comment by chinmaygarde** [comment]
Yeah, 1 just means no mipmapping.

**Review by chinmaygarde — APPROVED** [review]
Some nits but lgtm otherwise.

File: impeller/geometry/geometry_unittests.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/geometry/geometry_unittests.cc:643]: Done

File: impeller/geometry/size.h (hunk 1 above)
Comment (jonahwilliams) [reply impeller/geometry/size.h]: Done

**Pull request merged**